front/src/applications/stdcm/hooks/useLinkedPathSearch.ts

```diff
@@ -0,0 +1,170 @@
+import { useMemo, useState, useCallback } from 'react';
+
+import { groupBy } from 'lodash';
+import nextId from 'react-id-generator';
+import { useSelector } from 'react-redux';
+
+import type {
+  SearchPayload,
+  SearchResultItemOperationalPoint,
+  SearchResultItemTrainSchedule,
+} from 'common/api/osrdEditoastApi';
+import { osrdEditoastApi } from 'common/api/osrdEditoastApi';
+import { useOsrdConfActions, useOsrdConfSelectors } from 'common/osrdContext';
+import type { ExtremityPathStepType } from 'common/types';
+import type { StdcmConfSliceActions } from 'reducers/osrdconf/stdcmConf';
+import { useAppDispatch } from 'store';
+import { computeOpSchedules, isEqualDate } from 'utils/date';
+
+import { ArrivalTimeTypes, type StdcmLinkedPathResult, type StdcmLinkedPathStep } from '../types';
+
+type PathItemOperationalPoint = {
+  operational_point: string;
+  deleted?: boolean;
+  id: string;
+};
+
+const useLinkedPathSearch = () => {
+  const dispatch = useAppDispatch();
+  const [postSearch] = osrdEditoastApi.endpoints.postSearch.useMutation();
+
+  const { getTimetableID, getSearchDatetimeWindow } = useOsrdConfSelectors();
+  const { updateOrigin, updateDestination, updateAnteriorPath, updatePosteriorPath } =
+    useOsrdConfActions() as StdcmConfSliceActions;
+
+  const timetableID = useSelector(getTimetableID);
+  const searchDatetimeWindow = useSelector(getSearchDatetimeWindow);
+
+  const selectableSlot = useMemo(() => {
+    const startDate = searchDatetimeWindow ? new Date(searchDatetimeWindow.begin) : new Date();
+    startDate.setUTCHours(0, 0, 0);
+    return {
+      start: startDate,
+      end: searchDatetimeWindow?.end || startDate,
+    };
+  }, [searchDatetimeWindow]);
+
+  const [linkedPathDate, setLinkedPathDate] = useState<Date>(selectableSlot.start);
+  const [displaySearchButton, setDisplaySearchButton] = useState(true);
+  const [linkedPathID, setLinkedPathID] = useState<string>();
+  const [linkedPathResults, setLinkedPathResults] = useState<StdcmLinkedPathResult[]>([]);
+
+  const searchPayload = useMemo(
+    () =>
+      ({
+        object: 'trainschedule',
+        query: [
+          'and',
+          ['search', ['train_name'], linkedPathID],
+          ['=', ['timetable_id'], timetableID!],
+        ],
+      }) as SearchPayload,
+    [timetableID, linkedPathID]
+  );
+
+  const operationalPointSearch = useCallback(
```

Comment:
Maybe `getExtremitiesDetails` is more precise
```suggestion
  const getExtremitiesDetails = useCallback(
```